src/main/java/com/example/petbuddybackend/service/rating/RatingService.java

```diff
@@ -16,86 +15,118 @@
-            throw new IllegalActionException("User cannot rate himself");
-        }
-
-        return ratingMapper.mapToRatingResponse(
-                createOrUpdateRating(caretakerEmail, clientEmail, careId, rating, comment)
-        );
+        Rating ratingEntity = createOrUpdateRating(caretakerEmail, clientEmail, careId, rating, comment);
+        refreshRatingPhotos(ratingEntity);
+        return ratingMapper.mapToRatingResponse(ratingEntity);
     }
 
     public RatingResponse deleteRating(String clientEmail, Long careId) {
-        Care care = careService.getCareById(careId);
-        String caretakerEmail = care.getCaretaker().getEmail();
-
-        RatingKey ratingKey = new RatingKey(caretakerEmail, clientEmail, careId);
-        Rating rating = getRating(ratingKey);
-        ratingRepository.deleteById(ratingKey);
-
+        Rating rating = getRatingOfClient(careId, clientEmail);
+        ratingRepository.delete(rating);
+        refreshRatingPhotos(rating);
         return ratingMapper.mapToRatingResponse(rating);
     }
 
+    private Rating refreshRatingPhotos(Rating rating) {
+        userService.renewProfilePicture(rating.getCare().getClient().getAccountData());
+        return rating;
+    }
+
     private Rating createOrUpdateRating(String caretakerEmail, String clientEmail, Long careId, int rating,
                                         String comment) {
-        Rating ratingEntity = getOrCreateRating(caretakerEmail, clientEmail, careId);
+
+        Care care = careService.getCareById(careId);
+        assertNotRatingSelf(clientEmail, caretakerEmail);
+        assertCareOfClient(care, clientEmail);
+        assertPaidState(care);
+
+        Rating ratingEntity = ratingRepository.findById(careId).orElseGet(() ->
+                Rating.builder()
+                        .careId(care.getId())
+                        .care(care)
+                        .build()
+        );
 
         ratingEntity.setRating(rating);
         ratingEntity.setComment(comment);
 
         return ratingRepository.save(ratingEntity);
     }
 
-    private void assertCareBetweenCaretakerAndClientIsValid(String clientEmail, Care care) {
-
+    private void assertCareOfClient(Care care, String clientEmail) {
         if(!care.getClient().getEmail().equals(clientEmail)) {
-            throw new ForbiddenException("You are not client in this care");
+            throw new ForbiddenException(String.format(NOT_CLIENT_MESSAGE, clientEmail, care.getId()));
         }
+    }
 
-        if(!care.getCaretakerStatus().equals(CareStatus.PAID) || !care.getClientStatus().equals(CareStatus.PAID)) {
-            throw new IllegalActionException("Cannot rate unpaid care");
+    private void assertPaidState(Care care) {
+        CareStatus caretakerStatus = care.getCaretakerStatus();
+        CareStatus clientStatus = care.getClientStatus();
+
+        if(caretakerStatus != CareStatus.PAID || clientStatus != CareStatus.PAID) {
```

Comment:
ja bym tutaj zrobil juz tak jak powinno byc z nowymi statusami, czyli, ze opieka musi byc zamiast PAID to ACCEPTED i dodatkowo wtedy czas obecny musi byc wiekszy niz data rozpoczecia opieki

Reply:
Imo ten następny task jest od tego więc będę miał to z tyłu głowy ale na razie zostawiłbym

Reply:
No dobra